### PR TITLE
T32767 force superuser

### PR DESCRIPTION
**Pull request by manuq**
https://phabricator.endlessm.com/T32805

### CODE REVIEW

**Comment by manuq** [comment]
> I merged #8, so you'll need to rebase this on the current master. Happily, I think that change will make it easier to add a `REGULAR_USERS_CAN_MANAGE_CONTENT` option which could set the [`CAN_MANAGE_CONTENT`](https://github.com/learningequality/kolibri/blob/b87f9da274a6a9218419cf9f858be695aa04304a/kolibri/core/auth/constants/user_kinds.py) user kind over here:
> 
> https://github.com/endlessm/kolibri-desktop-auth-plugin/blob/main/kolibri_desktop_auth_plugin/models.py#L52-L54
> 
> (Either in addition to or instead of the current behaviour).

Great, using the CAN_MANAGE_CONTENT is way better. I have updated the PR to use it, and also renamed the setting to REGULAR_USERS_CAN_MANAGE_CONTENT. However after many testing all regular users are being created as superusers, regardless of the setting. So I tested from `main` branch and that is also the case. Unless I'm confused, I think that #8 has regressed the situation for regular users.

> I'm not quite sure what the relation is between a "user kind" and adding a role to a user, but it would give us behaviour slightly closer to what we are aiming for in some cases since it lets a user manage Kolibri content without being able to change other Kolibri settings :)

Yeah user kind and role kind is a bit confusing. Looking at the code it seems that role_kinds gets extended by the user_kinds. So facility.add_role should work with CAN_MANAGE_CONTENT

**Comment by dylanmccall** [comment]
Testing this on my end, this seemed to be working correctly (after building `org.learningequality.Kolibri with this change, since testing this stuff with the `.Devel` version is rather tedious), but when trying to create a regular user it's failing here:

```
Traceback (most recent call last):
  File "/app/lib/python3.8/site-packages/kolibri/dist/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/app/lib/python3.8/site-packages/kolibri/dist/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/app/lib/python3.8/site-packages/kolibri/dist/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/app/lib/python3.8/site-packages/kolibri/dist/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/app/lib/python3.8/site-packages/kolibri/dist/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/app/lib/python3.8/site-packages/kolibri_desktop_auth_plugin/views.py", line 12, in get
    user = authenticate(token=token)
  File "/app/lib/python3.8/site-packages/kolibri/dist/django/contrib/auth/__init__.py", line 70, in authenticate
    user = _authenticate_with_backend(backend, backend_path, request, credentials)
  File "/app/lib/python3.8/site-packages/kolibri/dist/django/contrib/auth/__init__.py", line 116, in _authenticate_with_backend
    return backend.authenticate(*args, **credentials)
  File "/app/lib/python3.8/site-packages/kolibri_desktop_auth_plugin/backends.py", line 45, in authenticate
    user = DesktopUser.objects.get_or_create(**user_details)
  File "/app/lib/python3.8/site-packages/kolibri_desktop_auth_plugin/models.py", line 21, in get_or_create
    try:
  File "/app/lib/python3.8/site-packages/kolibri_desktop_auth_plugin/models.py", line 62, in create_user
    DevicePermissions.objects.create(
  File "/app/lib/python3.8/site-packages/kolibri/dist/django/utils/functional.py", line 239, in inner
    return func(self._wrapped, *args)
  File "/app/lib/python3.8/site-packages/kolibri/dist/configobj.py", line 554, in __getitem__
    val = dict.__getitem__
```

Which is strange! I was thinking maybe the defaults weren't being added to the options dictionary here, but it looks like it just doesn't load. Might be something I'm doing wrong - i have a terrible headache today so haven't been able to look at it in any detail.

File: kolibri_desktop_auth_plugin/models.py [hunk 1]
```diff
@@ -61,8 +61,8 @@ def create_user(cls, uid, user_name, full_name=None, is_admin=False):
             )
         elif conf.OPTIONS["DesktopAuth"]["REGULAR_USERS_CAN_MANAGE_CONTENT"]:
             # allow regular user to manage content if the setting is enabled
-            kolibri_user.facility.add_role(
-                kolibri_user, user_kinds.CAN_MANAGE_CONTENT
+            DevicePermissions.objects.create(
+                user=kolibri_user, is_superuser=False, can_manage_content=True
```
Comment (manuq) [inline kolibri_desktop_auth_plugin/models.py:65]: :+1:  feel free to merge.

**Comment by dylanmccall** [comment]
For reference, the other issue here with the options dictionary is being addressed in learningequality/kolibri-installer-gnome#64.

**Pull request merged**